modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/model/DescriptorRegistry.java

```diff
@@ -50,10 +51,17 @@
     protected Map<String, Map<String, Map<String, List<Descriptor>>>> descriptors = new HashMap<>();
 
     public <T extends Descriptor> T getDescriptor(String target, String xp, String id) {
-        return (T) merge(descriptors.getOrDefault(target, Collections.emptyMap())
-                                    .getOrDefault(xp, Collections.emptyMap())
-                                    .getOrDefault(id, Collections.emptyList()));
+        return this.<T> optDescriptor(target, xp, id).orElse(null);
+    }
 
+    /**
+     * @since 11.4
+     */
+    public <T extends Descriptor> Optional<T> optDescriptor(String target, String xp, String id) {
```

Comment:
public or protected?

Reply:
I would keep the public modifier for later use if it's ok for the team?